### ISSUE TITLE
TSNone should not reset for TSVariable/TSError

### ISSUE THREAD

**Issue by theHamsta** [nvim-treesitter/nvim-treesitter#583]
**Describe the bug**
`TSVariable` and `TSError` is mapped to `TSNone` which may reset the foreground color even if it shouldn't.

**To Reproduce**
Just look at the C++ parser whether `&/*` pointer arguments are highlighted as parameters. Sometimes `TSParameter` wins against `TSNone` (C parser) and some times not (C++).  Another example are member variables in C++ `m_*`

`hlmap['variable'] = nil` solves the problem.

**Expected behavior**
`TSVariable` and `TSError` should be link to a highlight group with no directives (not `NONE` not `fg). They should not reset but just do nothing.

**Output of `:checkhealth nvim_treesitter`**

<details>
<code>

health#nvim_treesitter#check
========================================================================
## Installation
  - OK: `git` executable found.
  - OK: `cc` executable found.

## elm parser healthcheck
  - OK: elm parser found.
  - WARNING: No `highlights.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## wat parser healthcheck
  - OK: wat parser found.
  - WARNING: No `highlights.scm` query found for wat
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for wat
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for wat
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for wat
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## c parser healthcheck
  - OK: c parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## teal parser healthcheck
  - OK: teal parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for teal
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `folds.scm` found.

## java parser healthcheck
  - OK: java parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for java
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for java
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## python parser healthcheck
  - OK: python parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## dart parser healthcheck
  - OK: dart parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for dart
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for dart
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## lua parser healthcheck
  - OK: lua parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## ocaml parser healthcheck
  - OK: ocaml parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for ocaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `folds.scm` found.

## go parser healthcheck
  - OK: go parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - WARNING: No `folds.scm` query found for go
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## nix parser healthcheck
  - OK: nix parser found.
  - WARNING: No `highlights.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## yaml parser healthcheck
  - OK: yaml parser found.
  - WARNING: No `highlights.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## json parser healthcheck
  - OK: json parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for json
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for json
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for json
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## jsdoc parser healthcheck
  - OK: jsdoc parser found.
  - WARNING: No `highlights.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## php parser healthcheck
  - OK: php parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for php
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `folds.scm` found.

## julia parser healthcheck
  - OK: julia parser found.
  - WARNING: No `highlights.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## html parser healthcheck
  - OK: html parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for html
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for html
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## typescript parser healthcheck
  - OK: typescript parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## fennel parser healthcheck
  - OK: fennel parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for fennel
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for fennel
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## swift parser healthcheck
  - OK: swift parser found.
  - WARNING: No `highlights.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## ql parser healthcheck
  - OK: ql parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for ql
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for ql
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## query parser healthcheck
  - OK: query parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for query
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for query
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## javascript parser healthcheck
  - OK: javascript parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## cpp parser healthcheck
  - OK: cpp parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## regex parser healthcheck
  - OK: regex parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for regex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for regex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for regex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## kotlin parser healthcheck
  - OK: kotlin parser found.
  - WARNING: No `highlights.scm` query found for kotlin
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for kotlin
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for kotlin
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for kotlin
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## ruby parser healthcheck
  - OK: ruby parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for ruby
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for ruby
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## vue parser healthcheck
  - OK: vue parser found.
  - WARNING: No `highlights.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## ocamllex parser healthcheck
  - OK: ocamllex parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for ocamllex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for ocamllex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for ocamllex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## scala parser healthcheck
  - OK: scala parser found.
  - WARNING: No `highlights.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## toml parser healthcheck
  - OK: toml parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for toml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for toml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## rust parser healthcheck
  - OK: rust parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - WARNING: No `folds.scm` query found for rust
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## bash parser healthcheck
  - OK: bash parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for bash
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for bash
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## ocaml_interface parser healthcheck
  - OK: ocaml_interface parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for ocaml_interface
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `folds.scm` found.

## rst parser healthcheck
  - OK: rst parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - WARNING: No `folds.scm` query found for rst
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## css parser healthcheck
  - OK: css parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for css
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for css
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for css
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## zig parser healthcheck
  - OK: zig parser found.
  - WARNING: No `highlights.scm` query found for zig
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for zig
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for zig
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for zig
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## c_sharp parser healthcheck
  - OK: c_sharp parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for c_sharp
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `textobjects.scm` found.
  - WARNING: No `folds.scm` query found for c_sharp
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## haskell parser healthcheck
  - OK: haskell parser found.
  - WARNING: No `highlights.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## markdown parser healthcheck
  - OK: markdown parser found.
  - WARNING: No `highlights.scm` query found for markdown
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for markdown
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for markdown
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `folds.scm` found.

## tsx parser healthcheck
  - OK: tsx parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

</code>
</details>

**Output of `nvim --version`**
```
❯ nvim --version
NVIM v0.5.0-752-g759a05407
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: /usr/lib/ccache/clang-12 -fcolor-diagnostics -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/stephan/projects/neovim/release/config -I/home/stephan/projects/neovim/src -I/home/stephan/projects/neovim/.deps/usr/include -I/home/linuxbrew/.linuxbrew/Cellar/libuv/1.39.0/include -I/usr/include -I/home/stephan/projects/neovim/release/src/nvim/auto -I/home/stephan/projects/neovim/release/include
Compiled by stephan@stephan-Z87-DS3H

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/share/nvim"

Run :checkhealth for more info

```

**Additional context**

**Comment by kyazdani42** [comment]
none is supposed to do nothing, when we changed it to reset foreground and background, it might have gone out of its scope.
maybe an @reset would have been more appropriate to handle highlighting reset.

**Comment by theHamsta** [comment]
@kyazdani42  `NONE` is supposed to clear all previous highlights and it will do that. It has that meaning "reset" in the original vim usage. Maybe reset would be clearer.  I'd directly set the highlights for `TSVariable` `TSError` to nothing (empty string if vim accepts that).  People can still set it to some highlight group if they want to.

**Comment by theHamsta** [comment]
Fixed via #600
